src/rt/aaA.d

```diff
@@ -178,7 +236,7 @@ body
 
     if (aa.impl is null)
     {
-        aa.impl = new Impl();
+        aa.impl = new Impl(keyti, typeid(void*), valuesize);
```

Comment:
What's going on with the templated AA implementation? Would it hurt to add the value type information with #1037 and https://github.com/D-Programming-Language/dmd/pull/4159?